python/katana/util/buffer_access.pyx

```diff
@@ -0,0 +1,27 @@
+import numpy
+
+from cpython.buffer cimport PyBUF_CONTIG, PyBuffer_Release, PyObject_CheckBuffer, PyObject_GetBuffer
+
+
+cdef class UntypedBufferAccess:
+    def __init__(self, data):
+        # Handle pyarrow.Array
+        if not PyObject_CheckBuffer(data) and hasattr(data, "to_numpy"):
```

Comment:
why not move this check after the next if statement, then we do not need the first part of the condition